spec/features/tracing_spec.rb

```diff
@@ -1,7 +1,51 @@
 # frozen_string_literal: true
 
-RSpec.describe "Tracing", type: :feature do
-  it "works" do
+RSpec.describe "Tracing", type: :e2e do
+  def expect_valid_sample_rand(sample_rand)
+    expect(sample_rand).not_to be_nil
+    expect(sample_rand).to match(/^\d+\.\d{1,6}$/)
+    sample_rand_value = sample_rand.to_f
+    expect(sample_rand_value).to be >= 0.0
+    expect(sample_rand_value).to be < 1.0
+  end
+
+  def expect_dsc_in_envelope_headers
+    envelopes_with_dsc = logged_events[:envelopes].select do |envelope|
+      envelope["headers"] && envelope["headers"]["trace"]
+    end
+
+    expect(envelopes_with_dsc).not_to be_empty
+
+    dsc_metadata = envelopes_with_dsc.map { |envelope| envelope["headers"]["trace"] }
+
+    envelopes_with_sample_rand = dsc_metadata.select { |dsc| dsc["sample_rand"] }
+    expect(envelopes_with_sample_rand).not_to be_empty
+
+    envelopes_with_sample_rand.each do |dsc|
+      expect_valid_sample_rand(dsc["sample_rand"])
+    end
+
+    dsc_metadata
+  end
+
+  def get_http_server_transactions_with_headers
```

Comment:
sorry if I'm blind but where are we using this method?

Reply:
right, but then I wanted us to check the incoming request headers too by actually using this method, not removing it..

anyway it's just a spec so don't care that much, feel free to add it or not